solidity/dashboard/src/components/BondingSection.jsx

```diff
@@ -99,58 +100,41 @@ const AvailableEthCell = React.memo(({ availableETH }) => {
 
 const ActionCell = React.memo(
   ({ availableETH, availableETHInWei, operatorAddress, isWithdrawable }) => {
-    const { ModalComponent, openModal, closeModal } = useModal()
-
-    const web3Context = useWeb3Context()
-    const showMessage = useShowMessage()
-
-    const withdrawAllUnbondedAmount = useCallback(
-      async (onTransactionHashCallback) => {
-        try {
-          await tbtcAuthorizationService.withdrawAllEthForOperator(
-            web3Context,
-            { operatorAddress, availableETH },
-            onTransactionHashCallback
-          )
-          showMessage({
-            type: messageType.SUCCESS,
-            title: "Success",
-            content: "Withdrawal of ETH transaction has successfully completed",
-          })
-        } catch (error) {
-          showMessage({
-            type: messageType.ERROR,
-            title: "Withdrawal of ETH has failed ",
-            content: error.message,
-          })
-          throw error
-        }
-      },
-      [operatorAddress, availableETH, showMessage, web3Context]
-    )
+    const addEthModal = useModal()
```

Comment:
I think we should have just one instance of `Modal` and switch modal's content depending on the action. eg:

```
<ModalComponent title={title}>
 { action === "widthdraw" : <WithdrawEtH /> : <AddEth /> }
</ModalComponent>
```

or create a wrapper component that should render the right content depending on the props.

Reply:
Agree, changed. Thanks for the help!